pac4j-http/src/main/java/org/pac4j/http/credentials/authenticator/RestAuthenticator.java

```diff
@@ -0,0 +1,142 @@
+        mapper.configure(JsonParser.Feature.ALLOW_SINGLE_QUOTES, true);
+    }
+
+    private String url;
+
+    public RestAuthenticator() {}
+
+    public RestAuthenticator(final String url) {
+        this.url = url;
+    }
+
+    @Override
+    protected void internalInit(final WebContext context) {
+        CommonHelper.assertNotBlank("url", url);
+        defaultProfileDefinition(new CommonProfileDefinition<>(x -> new RestProfile()));
+    }
+
+    @Override
+    public void validate(final UsernamePasswordCredentials credentials, final WebContext context) throws HttpAction, CredentialsException {
+        init(context);
+
+        final String username = credentials.getUsername();
+        final String password = credentials.getPassword();
+        if (CommonHelper.isBlank(username) || CommonHelper.isBlank(password)) {
+            logger.error("Empty username or password");
+            return;
+        }
+
+        final String body = callRestApi(username, password);
+        logger.debug("body: {}", body);
+        if (body != null) {
+            buildProfile(credentials, body);
+        }
+    }
+
+    protected void buildProfile(final UsernamePasswordCredentials credentials, final String body) throws HttpAction, CredentialsException {
+        final RestProfile profileClass = getProfileDefinition().newProfile();
+        final RestProfile profile;
+        try {
+            profile = mapper.readValue(body, profileClass.getClass());
+        } catch (final IOException e) {
+            throw new TechnicalException(e);
+        }
+        logger.debug("profile: {}", profile);
+        credentials.setUserProfile(profile);
+    }
+
+    /**
+     * Return the body from the REST API, passing the username/pasword auth.
+     * To be overridden using another HTTP client if necessary.
+     *
+     * @param username the username
+     * @param password the password
+     * @return the response body
+     * @throws HttpAction whether an extra HTTP action is required
+     * @throws CredentialsException whether an authentication error occurs
+     */
+    protected String callRestApi(final String username, final String password) throws HttpAction, CredentialsException {
+
+        final String basicAuth;
+        try {
+            basicAuth = Base64.getEncoder().encodeToString((username + ":" + password).getBytes("UTF-8"));
+        } catch (final UnsupportedEncodingException e) {
+            throw new TechnicalException(e);
+        }
+
+        final Map<String, String> headers = new HashMap<>();
+        headers.put(HttpConstants.AUTHORIZATION_HEADER, HttpConstants.BASIC_HEADER_PREFIX + basicAuth);
+
+        HttpURLConnection connection = null;
+        try {
+            connection = HttpUtils.openPostConnection(new URL(url), headers);
+            int code = connection.getResponseCode();
+            if (code != 200) {
+                logger.error("Failed response: {}", HttpUtils.buildHttpErrorMessage(connection));
```

Comment:
idem with the error log: is that really a good idea to log everytime a user can't authenticate against the REST service?